### PR TITLE
DropDownList based editors: dropDownOptions.width should be passed to the popup correctly (T897820)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/ui/date_box/ui.date_box.base.js [hunk 1]
```diff
@@ -259,6 +259,7 @@ const DateBox = DropDownEditor.inherit({
     _renderDimensions: function() {
         this.callBase();
         this.$element().toggleClass(DX_AUTO_WIDTH_CLASS, !this.option('width'));
+        this._strategy && this._strategy._dimensionChanged();
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.base.js]:
```suggestion
        this._strategy?._dimensionChanged();
```

File: js/ui/date_box/ui.date_box.strategy.js [hunk 2]
```diff
@@ -57,6 +57,11 @@ const DateBoxStrategy = Class.inherit({
 
     popupConfig: abstract,
 
+    _dimensionChanged: function() {
+        const popup = this._getPopup();
+        popup && popup.repaint();
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.strategy.js]:
```suggestion
        this._getPopup()?.repaint();
```

File: js/ui/date_box/ui.date_box.strategy.list.js [hunk 3]
```diff
@@ -256,16 +250,7 @@ const ListStrategy = DateBoxStrategy.inherit({
...
-    },
-
-    _updatePopupDimensions: function() {
-        this._updatePopupWidth();
-        this._updatePopupHeight();
-    },
-
-    _updatePopupWidth: function() {
-        this.dateBox._setPopupOption('width', this._getPopupWidth());
+        this._getPopup() && this._updatePopupHeight();
```
Comment (DokaRus) [inline js/ui/date_box/ui.date_box.strategy.list.js]:
```suggestion
        this._getPopup()?._updatePopupHeight();
```

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js [hunk 4]
```diff
@@ -94,6 +94,23 @@ QUnit.module('Color Box', {
         fx.off = false;
     }
 }, () => {
+    QUnit.skip('popup should have correct width after editor width runtime change', function(assert) {
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js]: Should we remove it?

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js [hunk 5]
```diff
@@ -2009,6 +2009,23 @@ QUnit.module('popup options', {
         this.clock.restore();
     }
 }, () => {
+    QUnit.skip('popup should have correct width after editor width runtime change', function(assert) {
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js]: Should we remove it?

File: testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js [hunk 6]
```diff
@@ -74,6 +74,23 @@ QUnit.module('popup integration', {
         this.popup = getPopup(this.instance);
     }
 }, () => {
+    QUnit.skip('popup should have correct width after editor width runtime change', function(assert) {
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js]: Should we remove it?

File: testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js (hunk 4 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/colorBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js (hunk 5 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js (hunk 6 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets/dropDownButton.tests.js]: fixed

File: js/ui/date_box/ui.date_box.base.js (hunk 1 above)
Comment (ksercs) [reply js/ui/date_box/ui.date_box.base.js]: fixed

File: js/ui/date_box/ui.date_box.strategy.js (hunk 2 above)
Comment (ksercs) [reply js/ui/date_box/ui.date_box.strategy.js]: fixed

File: js/ui/date_box/ui.date_box.strategy.list.js (hunk 3 above)
Comment (ksercs) [reply js/ui/date_box/ui.date_box.strategy.list.js]: fixed

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**